packages/ckeditor5-language/tests/textfragmentlanguagecommand.js

```diff
@@ -0,0 +1,385 @@
+
+			command.execute( { languageCode: 'ar', textDirection: 'rtl' } );
+
+			expect( command.value ).to.equal( 'ar:rtl' );
+			expect( doc.selection.getAttribute( 'language' ) ).to.equal( 'ar:rtl' );
+
+			command.execute( { languageCode: false } );
+
+			expect( command.value ).to.be.false;
+			expect( doc.selection.hasAttribute( 'language' ) ).to.be.false;
+		} );
+
+		it( 'should not store attribute change on selection if selection is collapsed in non-empty parent', () => {
+			setData( model, '<p>a[]bc<$text language="fr:ltr">foobar</$text>xyz</p>' );
+
+			command.execute( { languageCode: 'ar', textDirection: 'rtl' } );
+
+			// It should not save that language was executed at position ( root, [ 0, 1 ] ).
+
+			model.change( writer => {
+				// Simulate clicking right arrow key by changing selection ranges.
+				writer.setSelection( writer.createRange(
+					writer.createPositionAt( root.getNodeByPath( [ 0 ] ), 2 )
+				) );
+
+				// Get back to previous selection.
+				writer.setSelection( writer.createRange(
+					writer.createPositionAt( root.getNodeByPath( [ 0 ] ), 1 )
+				) );
+			} );
+
+			expect( command.value ).to.be.false;
+		} );
+
+		it( 'should change selection attribute and store it if selection is collapsed in empty parent', () => {
+			setData( model, '<p>abc<$text language="fr:ltr">foobar</$text>xyz</p><p>[]</p>' );
+
+			expect( command.value ).to.be.false;
+
+			command.execute( { languageCode: 'ar', textDirection: 'rtl' } );
+
+			expect( command.value ).to.equal( 'ar:rtl' );
+			expect( doc.selection.getAttribute( 'language' ) ).to.equal( 'ar:rtl' );
+
+			// Attribute should be stored.
+			// Simulate clicking somewhere else in the editor.
+			model.change( writer => {
+				writer.setSelection( root.getNodeByPath( [ 0 ] ), 2 );
+			} );
+
+			expect( command.value ).to.be.false;
+
+			// Go back to where attribute was stored.
+			model.change( writer => {
+				writer.setSelection( root.getNodeByPath( [ 1 ] ), 0 );
+			} );
+
+			// Attribute should be restored.
+			expect( command.value ).to.equal( 'ar:rtl' );
+
+			command.execute( { languageCode: false } );
+
+			expect( command.value ).to.be.false;
+			expect( doc.selection.hasAttribute( 'language' ) ).to.be.false;
+		} );
+
+		it( 'should force language text direction if textDirection was set', () => {
+			setData( model, '<p>x[]y</p>' );
+
+			command.execute( { languageCode: 'ar', textDirection: 'ltr' } );
+
+			expect( command.value ).to.equal( 'ar:ltr' );
+			expect( doc.selection.getAttribute( 'language' ) ).to.equal( 'ar:ltr' );
+		} );
+
+		it( 'should detect language text direction if textDirection was not set', () => {
+			setData( model, '<p>x[]y</p>' );
+
+			command.execute( { languageCode: 'ar' } );
+
+			expect( command.value ).to.equal( 'ar:rtl' );
+			expect( doc.selection.getAttribute( 'language' ) ).to.equal( 'ar:rtl' );
+		} );
+
+		describe( 'model change event', () => {
+			let spy;
+
+			beforeEach( () => {
+				spy = sinon.spy();
```

Comment:
We should init sinon sandbox before using it (`testUtils.createSinonSandbox();`)